lib/timeline/component/ItemSet.js

```diff
@@ -634,9 +634,14 @@ ItemSet.prototype.redraw = function() {
   var visibleInterval = range.end - range.start;
   var zoomed = (visibleInterval != this.lastVisibleInterval) || (this.props.width != this.props.lastWidth);
   var scrolled = range.start != this.lastRangeStart;
-  var forceRestack = (zoomed || scrolled);
+  var changedStack = options.stack != this.lastStack
+  var changedStackSubgroups = options.stackSubgroups != this.lastStackSubgroups
+  var forceRestack = (zoomed || scrolled || changedStack || changedStackSubgroups);
   this.lastVisibleInterval = visibleInterval;
   this.lastRangeStart = range.start;
+  this.lastStack = options.stack;
```

Comment:
The logic is good, but the `lastStack` would be better as `lastStackOptions` - its not really the "last stacking"